### PR TITLE
Lps 54048 As a developer, I would like to make cluster channel handle its address.

### PR DESCRIPTION
**Pull request by tinatian**

### CODE REVIEW

**Comment by shuyangzhou** [comment]
@tinatian about https://github.com/tinatian/liferay-portal/commit/c7e49ebf34ce9058c4844d914044b2038c33b09d

Having ClusterNode to manage Addresses is fine. But do we really want the addresses to be part of the equals()/hashCode() logic of ClusterNode?

1) The address info are populated after constructor call, so technically, the equals()/hashCode() results are dynamic changing during the populating process.
2) When jgroups disconnect/reconnect, will the address change? If it does, then it could break the ClusterNode's consumers.

The bottom line, I think we created ClusterNode to be a logical representation for a portal instance in cluster. It should not identify itself with the lower level physical info.

**Comment by tinatian** [comment]
I think the root argument comes from there are two different meanings about SAME cluster node:
a) are they from same portal instance
b) is current cluster structure changed

Which one should be implemented in the equals method of ClusterNode? 
I chose b)
 
Becasue having ClusterNode to manage Addresses means ClusterNode is responsible for the structure of cluster, then ClusterNode should represent the changes in cluster structure, that is to compare Addresses in equals.

Aslo, suppose we removed the comparison of Addresses from equals, we are comparing all the attributes of ClusterNode except Addresses, isn't that weird?

For the effect on ClusterNode's consummers you are talking about, I think consummer should know which kind of SAME they are meaning and choose the right way to do the comparson. While, it is true that it adds more work to consummers, I did not figure out a more reasonable way to do this.

**Comment by shuyangzhou** [comment]
We need both ways of "same cluster node definition". Purely from logic point of view, they will be both needed eventually.

I trend to think ClusterNode as a logical definition of cluster node, plus the additional attached addresses, it can help to serve the physical definition of cluster. To get that, the clusternode can not include addresses for equals/hashCode.

From API point of view, for either case, we should not expose addChannelAddress() on ClusterNode, as it is never a valid method to be called by client, then it should not be exposed. We need to at least hide this. Same rules should be applied to every other methods that changes ClusterNode's internal states. If client should not call it, we should not expose it.

If you want to keep ClusterNode to only represents physical cluster, that's fine, but you need to also provide a way to represents logical one. We have to cover them both.

**Comment by shuyangzhou** [comment]
I merged in the first 2 commits. See https://github.com/brianchandotcom/liferay-portal/pull/24851

**Comment by tinatian** [comment]
I think it's better to put Addresses in another place instead of cluster node, then we should not need to worry all these visibilty issues.

Since current cluster logic is broken, I can not test the changes, will send you a new PR later.
